messaging/broker.go

```diff
@@ -928,7 +928,7 @@ func ReadSegmentByIndex(path string, index uint64) (*Segment, error) {
 	} else if index == 0 {
 		return segments[0], nil
 	} else if index < segments[0].Index {
-		return nil, ErrSegmentReclaimed
```

Comment:
We don't have truncation in yet so we're not tombstoning so the reclamation error causes problems when we materialize the broker and have no topic data. Once we have tombstones in then we'll be able to check against that.